pkg/engine/background.go

```diff
@@ -69,12 +70,30 @@ func filterRule(rclient registryclient.Client, rule kyvernov1.Rule, policyContex
 		return nil
 	}
 
+	// check if there is a corresponding policy exception
+	exception, err := matchesException(policyContext, &rule)
+	// if we found an exception
+	if err == nil && exception != nil {
```

Comment:
can we make the handling of the exception reusable for all the rule types? Currently we are duplicating the same 15-20 lines of code in each.

Reply:
I extracted the logic to a separate function `shouldRespondException`.